src/tests/multihost/admultidomain/test_multidomain.py

```diff
@@ -0,0 +1,187 @@
+""" AD-Provider AD Parameters tests ported from bash
+
+:requirement: ad_parameters
+:casecomponent: sssd
+:subsystemteam: sst_idm_sssd
+:upstream: yes
+"""
+import tempfile
+import pytest
+
+from sssd.testlib.common.utils import sssdTools
+from sssd.testlib.common.utils import SSSDException
+from sssd.testlib.common.utils import ADOperations
+
+
+@pytest.fixture(scope="class")
+def change_client_hostname(session_multihost, request):
+    """ Change client hostname to a truncated version in the AD domain"""
+    cmd = session_multihost.client[0].run_command('hostname', raiseonerr=False)
+    old_hostname = cmd.stdout_text.rstrip()
+    ad_domain = session_multihost.ad[0].domainname
+    try:
+        new_hostname = session_multihost.client[0].external_hostname. \
+            split('.')[0]
+    except (KeyError, AttributeError):
+        new_hostname = old_hostname.split('.')[0]
+    if new_hostname.startswith('ci-'):
+        new_hostname = new_hostname[3:]
+    new_hostname = new_hostname[:15] + "." + ad_domain
+    session_multihost.client[0].run_command(
+        f'hostname {new_hostname}', raiseonerr=False
+    )
+
+    def restore():
+        """ Restore hostname """
+        session_multihost.client[0].run_command(
+            f'hostname {old_hostname}',
+            raiseonerr=False
+        )
+    request.addfinalizer(restore)
+
+
+def ssh_setup(session_multihost, user, group=""):
+    """Setup a ssh key for a root and add it to authorized keys for user
+    :param group: group used to set file ownership
+    :param session_multihost: multihost object
+    :param user: username including domain if needed
+    :returns whether command succeeded
+    """
+    with tempfile.NamedTemporaryFile(mode='w') as tfile:
```

Comment:
Is this used anywhere?

If you're running all this on the remove host, why not simply use multiple run_commands instead of creating a temp script?   If something fails in the script, you'll have less information about what step it actually failed on for troubleshooting won't you?

You can make this a little more readable by loading the entire contents into a var and using put_file_contents instead of put_file.  Something like:

```
    sshd_cont = 'first line\n' \
                'second line\n' \
                'last line\n'
    session_multihost.client[0].transport.put_file_contents('/tmp/ssh_setup.sh', sshd_cont)
```

Reply:
This was somebody else code that was compiled over. It's actually not used at all, so I removed it.